plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -578,12 +586,20 @@ long getRetainedSizeInBytes()
         }
     }
 
-    private record QueriedTable(String queryId, String tableLocation, long version)
+    private record QueriedLocation(String queryId, String tableLocation)
     {
-        QueriedTable
+        QueriedLocation
         {
             requireNonNull(queryId, "queryId is null");
             requireNonNull(tableLocation, "tableLocation is null");
         }
     }
+
+    private record QueriedTable(QueriedLocation queriedLocation, long version)
```

Comment:
Can we cache just per-version (without queryID). Now if we have two queries reading same table at same version we will have two entries in cache. Right?

Reply:
they will use same snapshot, via `TransactionLogAccess.tableSnapshots` cache